pkg/component/runtime/runtime.go

```diff
@@ -176,7 +176,10 @@ func (s *componentRuntimeState) destroy() {
 	if s.watchCanceller != nil {
 		s.watchCanceller()
 		s.watchCanceller = nil
-		<-s.watchChan
+		// Do not wait on watchChan here
+		// the watch loop calls stateChanged that calls "destroy" (this function) and will block forever
+		// since the watch channel will never be closed because the watch loop is blocked on stateChanged call
+		// <-s.watchChan
```

Comment:
I think we need to ensure that we wait for the watcher goroutine does stop. This seems like a hacky way of fixing it, by just commenting it out.

Maybe we could use a different channel to know when its done? Or a waitgroup?

Reply:
I'll rewrite how the runners are managed, it currently deadlocks on wait for this channel to be closed while it can't be closed until you return from this function.

Reply:
this is rewritten now, think it's cleaner